### ISSUE TITLE
How do I load data from a csv file

### ISSUE THREAD

**Issue by nawshad** [pytorch/text#711]
I have a dataset containing text and labels seperated by tabs. How I can load this dataset using torchtext?

**Comment by zhangguanheng66** [comment]
It's probably similar to the text classification datasets [here](https://github.com/pytorch/text/blob/master/torchtext/datasets/text_classification.py).

@mttk Do you know if the current library support a csv file loading?

**Comment by bentrevett** [comment]
```python
from torchtext import data

TEXT = data.Field()
LABEL = data.LabelField()

fields = [('text', TEXT), ('label', LABEL)]

train_data, test_data = data.TabularDataset.splits(
                            path = 'data',
                            train = 'train.tsv',
                            test = 'test.tsv',
                            format = 'tsv',
                            fields = fields
)
```

**Comment by M-e-r-c-u-r-y** [comment]
@zhangguanheng66 Could we add the example @bentrevett posted in an example/usage section, torchtext doesn't really have any examples for external data sets. Adding a few examples for datasets that are not built into torchtext will help new users in understanding how to use torchtext better.

**Comment by zhangguanheng66** [comment]
We plan to eventually retire `Field` class as legacy code. However, at this moment, we could land a OSS PR as the example to help the usage case above. @M-e-r-c-u-r-y

**Comment by GaoJiqiang** [comment]
How can I load AG_news or DBpedia datasets from local csv file using 'text_classification.DATASETS' instead of from google drive?

**Comment by zhangguanheng66** [comment]
If you have the paths of train/test files for AG_NEWS and DBpedia, you could save them as a list and start from [here](https://github.com/pytorch/text/blob/43acc7534738ca6af91c19524667738d4b3f1fa3/torchtext/datasets/text_classification.py#L118). So why not just call the AG_NEWS and DBpedia API to load the datasets?

**Comment by GaoJiqiang** [comment]
> If you have the paths of train/test files for AG_NEWS and DBpedia, you could save them as a list and start from [here](https://github.com/pytorch/text/blob/43acc7534738ca6af91c19524667738d4b3f1fa3/torchtext/datasets/text_classification.py#L118). So why not just call the AG_NEWS and DBpedia API to load the datasets?

Thx, because every time I have to use vpn to run the project for getting data from google drive, it's a little trouble. I want to download the CSV file and store them in local directory for convenience. I will try your method, best wishes.

**Comment by y12uc231** [comment]
Is there a way to load datasets from CSV files in torchtext == 0.12? It seems like they removed legacy as well.

**Comment by parmeet** [comment]
@y12uc231  In torchtext 0.12 we have migrated our datasets on top of [torchdata](https://github.com/pytorch/data). You can look at datasets implementation that offer plenty of examples how to work with CSV files or refer the torchdata [documentation](https://pytorch.org/data/beta/index.html) for additional information on usage and available functionality in datapipes.

Datapipe for reading data from CSV files is [here](https://pytorch.org/data/beta/generated/torchdata.datapipes.iter.CSVParser.html#torchdata.datapipes.iter.CSVParser)

```python
from torchdata.datapipes.iter import IterableWrapper, FileOpener
dp = IterableWrapper(["my_csv_file.csv"])
dp = FileOpener(dp, mode='b')
dp = dp.parse_csv()

for sample in dp:
     print(sample)
```

**Comment by y12uc231** [comment]
Thanks! This works!

Maybe a bit of a different question but do you know how to load Glove embedding vocabulary for my dataset?
